lib/src/command/lish.dart

```diff
@@ -169,11 +165,11 @@ class LishCommand extends PubCommand {
       } else {
         // For third party servers using bearer authentication client
         await withAuthenticatedClient(cache, server, (client) {
-          return _publishUsingClient(packageBytes, client);
+          return _publishUsingClient(packageBytes, client as http.BaseClient);
```

Comment:
It seems to me that `_publishUsingClient` should take a `http.Client` and you would not need the cast.